pkgs/development/libraries/raylib/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchFromGitHub, cmake,
+  mesa, libGLU, glfw,
+  libX11, libXi, libXcursor, libXrandr, libXinerama,
+  alsaSupport ? stdenv.hostPlatform.isLinux, alsaLib,
+  pulseSupport ? stdenv.hostPlatform.isLinux, libpulseaudio,
+  includeEverything ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "raylib";
+  version = "3.5.0";
+
+  src = fetchFromGitHub {
+    owner = "raysan5";
+    repo = pname;
+    rev = version;
+    sha256 = "0syvd5js1lbx3g4cddwwncqg95l6hb3fdz5nsh5pqy7fr6v84kwj";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    mesa libGLU glfw libX11 libXi libXcursor libXrandr libXinerama
+  ] ++ lib.optional alsaSupport alsaLib
+    ++ lib.optional pulseSupport libpulseaudio;
+
+  # https://github.com/raysan5/raylib/wiki/CMake-Build-Options
+  cmakeFlags = [
+    "-DUSE_EXTERNAL_GLFW=ON"
+    "-DSHARED=ON"
+    "-DBUILD_EXAMPLES=OFF"
+  ] ++ lib.optional includeEverything "-DINCLUDE_EVERYTHING=ON";
```

Comment:
What does `includeEverything` do?

Reply:
Enables all the optional image codecs.

https://github.com/raysan5/raylib/wiki/CMake-Build-Options

Binary Option | Description | Default Value
-- | -- | --
INCLUDE_EVERYTHING | Include all optional off-by-default formats in build | OFF
SUPPORT_FILEFORMAT_PNG | Support loading PNG as textures | ON
SUPPORT_FILEFORMAT_DDS | Support loading DDS as textures | ON
SUPPORT_FILEFORMAT_HDR | Support loading HDR as textures | ON
SUPPORT_FILEFORMAT_KTX | Support loading KTX as textures | ON
SUPPORT_FILEFORMAT_ASTC | Support loading ASTC as textures | ON
SUPPORT_FILEFORMAT_BMP | Support loading BMP as textures | OFF
SUPPORT_FILEFORMAT_TGA | Support loading TGA as textures | OFF
SUPPORT_FILEFORMAT_JPG | Support loading JPG as textures | OFF
SUPPORT_FILEFORMAT_GIF | Support loading GIF as textures | OFF
SUPPORT_FILEFORMAT_PSD | Support loading PSD as textures | OFF
SUPPORT_FILEFORMAT_PKM | Support loading PKM as textures | OFF
SUPPORT_FILEFORMAT_PVR | Support loading PVR as textures | OFF
SUPPORT_FILEFORMAT_OBJ | Support loading OBJ file format | ON
SUPPORT_FILEFORMAT_MTL | Support loading MTL file format | ON

Reply:
They are auto-detected in the `cmakeConfigurePhase`?

Reply:
These defines enable support for writing to stdout the various image file formats. Relevant code is here: [stb_image_write.h](https://github.com/raysan5/raylib/blob/82d61a5875c4a7eaa4c8caf9ed3b19a8441eb258/src/external/stb_image_write.h).

[stb_image.h](https://github.com/raysan5/raylib/blob/82d61a5875c4a7eaa4c8caf9ed3b19a8441eb258/src/external/stb_image.h) handles the reading of various images and always supports those file formats.

There are no extra dependencies to enable this functionality.